### ISSUE TITLE
Setting OptionCheckSyntax to false throws a NullReferenceException if html has unclosed tags

### ISSUE THREAD

**Issue by Shoh** [zzzprojects/html-agility-pack#329]
### Description
When using the `LoadHtml` method of an `HtmlDocument` instance that has the property `OptionCheckSyntax` set to `false`, a `NullReferenceException` is thrown if the input HTML has any unclosed tags or is invalid in any way.  I'm not sure if this is expected or As Designed but I couldn't find any information about it.

### Exception
```
Exception message:  Unhandled Exception: System.NullReferenceException: Object reference not set to an instance of an object.
Stack trace:
   at HtmlAgilityPack.HtmlNode.UpdateLastNode() in C:\Users\Jonathan\source\repos\HtmlAgilityPack\HtmlAgilityPack.Shared\HtmlNode.cs:line 2009
   at HtmlAgilityPack.HtmlNode.CloseNode(HtmlNode endnode, Int32 level) in C:\Users\Jonathan\source\repos\HtmlAgilityPack\HtmlAgilityPack.Shared\HtmlNode.cs:line 2077
   at HtmlAgilityPack.HtmlNode.CloseNode(HtmlNode endnode, Int32 level) in C:\Users\Jonathan\source\repos\HtmlAgilityPack\HtmlAgilityPack.Shared\HtmlNode.cs:line 2056
   at HtmlAgilityPack.HtmlDocument.CloseCurrentNode() in C:\Users\Jonathan\source\repos\HtmlAgilityPack\HtmlAgilityPack.Shared\HtmlDocument.cs:line 1077
   at HtmlAgilityPack.HtmlDocument.PushNodeEnd(Int32 index, Boolean close) in C:\Users\Jonathan\source\repos\HtmlAgilityPack\HtmlAgilityPack.Shared\HtmlDocument.cs:line 2061
   at HtmlAgilityPack.HtmlDocument.Parse() in C:\Users\Jonathan\source\repos\HtmlAgilityPack\HtmlAgilityPack.Shared\HtmlDocument.cs:line 1391
   at HtmlAgilityPack.HtmlDocument.Load(TextReader reader) in C:\Users\Jonathan\source\repos\HtmlAgilityPack\HtmlAgilityPack.Shared\HtmlDocument.cs:line 771
   at HtmlAgilityPack.HtmlDocument.LoadHtml(String html) in C:\Users\Jonathan\source\repos\HtmlAgilityPack\HtmlAgilityPack.Shared\HtmlDocument.cs:line 819
   at Program.Main()
```

### Fiddle or Project
https://dotnetfiddle.net/9NZotJ

### Further technical details
- HAP version:  1.11.12
- NET version:  netcoreapp2.2

**Comment by JonathanMagnan** [comment]
Hello @Shoh ,

Thank you for reporting, we will look at why this issue happens.

Best Regards,

Jonathan

---

**Performance Libraries**
`context.BulkInsert(list, options => options.BatchSize = 1000);  `
[Entity Framework Extensions](http://entityframework-extensions.net/) • [Entity Framework Classic](http://entityframework-classic.net/) • [Bulk Operations](http://bulk-operations.net/) •  [Dapper Plus](http://dapper-plus.net/)

**Runtime Evaluation**
`Eval.Execute("x + y", new {x = 1, y = 2}); // return 3`
[C# Eval Function](http://eval-expression.net/) • [SQL Eval Function](http://eval-sql.net/)

**Comment by JonathanMagnan** [comment]
Hello @Shoh ,

The pull made by @EvgeniyZ has been finally merged. Thank to him!

Starting from the v1.11.15, you should no longer get the issue.

Let me know if everything works as expected.

Best Regards,

Jonathan

**Comment by JonathanMagnan** [comment]
Hello @Shoh 

Did you get the time to try the latest version v1.11.15?

Best Regards,

Jonathan

**Comment by Shoh** [comment]
Hi @JonathanMagnan,

Sorry for the delay; I've been on vacation.  Yes, I can confirm that the issue is fixed in v1.11.15.  Thank you!

-Shoh
